### ISSUE TITLE
"The specified module could not be found" when using windows_eventlog2 

### ISSUE THREAD

**Issue by flurreN** [fluent-plugins-nursery/td-agent-builder#98]
Everything works when using the [example windows_eventlog](https://github.com/fluent/fluent-plugin-windows-eventlog#in_windows_eventlog) config but not when im using my own or the [example windows_eventlog2](https://github.com/fluent/fluent-plugin-windows-eventlog#in_windows_eventlog2) config.

plugin versions:
```
winevt_c' 0.8.1
fluent-plugin-windows-eventlog 0.7.0
```
Tried on Windows 10 and Windows Server 2019 

Errorlog:
```
C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require': 126: The specified module could not be found.   - C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/winevt_c-0.8.1/lib/winevt/winevt.so (LoadError)
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/winevt_c-0.8.1/lib/winevt.rb:4:in `rescue in <top (required)>'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/winevt_c-0.8.1/lib/winevt.rb:1:in `<top (required)>'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:133:in `require'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:133:in `rescue in require'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:39:in `require'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluent-plugin-windows-eventlog-0.7.0/lib/fluent/plugin/in_windows_eventlog2.rb:1:in `<top (required)>'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/registry.rb:102:in `block in search'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/registry.rb:99:in `each'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/registry.rb:99:in `search'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/registry.rb:44:in `lookup'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/plugin.rb:155:in `new_impl'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/plugin.rb:105:in `new_input'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/root_agent.rb:312:in `add_source'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/root_agent.rb:158:in `block in configure'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/root_agent.rb:152:in `each'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/root_agent.rb:152:in `configure'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/engine.rb:105:in `configure'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/engine.rb:80:in `run_configure'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/supervisor.rb:551:in `run_supervisor'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/lib/fluent/command/fluentd.rb:330:in `<top (required)>'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/opt/td-agent/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/opt/td-agent/lib/ruby/gems/2.4.0/gems/fluentd-1.10.2/bin/fluentd:8:in `<top (required)>'
        from C:/opt/td-agent/bin/fluentd:23:in `load'
        from C:/opt/td-agent/bin/fluentd:23:in `<main>'
```

**Comment by cosmo0920** [comment]
Could you share your modification for td-agent-builder?
We're not using winevt_c v0.8.1 and fluent-plugin-windows-eventlog v0.7.0 in td-agent-builder for now.

**Comment by cosmo0920** [comment]
I'd confirmed that omnibus version can handle winevt_c v0.8.1 fat gem. But td-agent-builder version does not handle winevt_c v0.8.1 fat gem....
This should be suggested some breakage occurring on td-agent-builder version installation....

**Comment by ashie** [comment]
> I'd confirmed that omnibus version can handle winevt_c v0.8.1 fat gem. But td-agent-builder version does not handle winevt_c v0.8.1 fat gem....

This issue is only for fat gem?
I can't reproduce it when I use bundled winevt_c or build from the source.

**Comment by cosmo0920** [comment]
This issue also happens on msi installer installation.

**Comment by ashie** [comment]
I heard from @cosmo0920 that we should use clean environment instead of development environment to reproduce this issue. I'll try it.

**Comment by ashie** [comment]
Probably this is because that the installer doesn't bundle msys2 libraries.

**Comment by ashie** [comment]
```
x64-msvcrt-ruby240.dll
ADVAPI32.dll
KERNEL32.dll
msvcrt.dll
ole32.dll
wevtapi.dll
libgcc_s_seh-1.dll
libssp-0.dll
libstdc++-6.dll
```

**Comment by cosmo0920** [comment]
Ah, I got it.
Copying from libstdc++6.dll on C:\msys64\mingw64\bin to Ruby executable path is works for me.
libgcc_s_seh-1.dll, libssp-0.dll, and x64-msvcrt-ruby240.dll should be bundled in RubyInstaller and ADVAPI32.dllm, KERNEL32.dll, msvcrt.dll, ole32.dll, and wevtapi.dl also should be built-in dlls.

**Comment by flurreN** [comment]
> Could you share your modification for td-agent-builder?
> We're not using winevt_c v0.8.1 and fluent-plugin-windows-eventlog v0.7.0 in td-agent-builder for now.

I only changed this file for those two plugins https://github.com/clear-code/td-agent-builder/blob/master/td-agent/plugin_gems.rb , but I guess you already onto the problem!


Thanks for looking into this!

**Comment by ashie** [comment]
It was fixed via #103.
